### DISCUSSION TITLE
Run failed when constructing a pregnancy model

### DISCUSSION THREAD

**Discussion by D7abc** [Open-Systems-Pharmacology/Forum#1654] [Questions & Problems]
hello，The required parameters have been filled in, and the steps follow the tutorial. If run failed appears, what should I do

![Uploading YBAP%Q_HFRL9O4P$A{G$TMW.jpg…]()

**Comment by StephanSchaller** [comment]
@D7abc , which tutorial are you referring to?

**Reply by D7abc** [reply]
I refer to Dallmann's article：Applied Concepts in PBPK Modeling: How to Extend an Open Systems Pharmacology Model to the Special Population of Pregnant Women，Then “pregnantwoman” added the parameter 'total hectocrit', and added 'fup fetus' in' modules'
[教程.pdf](https://github.com/Open-Systems-Pharmacology/Forum/files/13280612/default.pdf)

**Reply by AndreDlm** [reply]
What's the error message that you get?

**Reply by D7abc** [reply]
Sorry for replying to you so late, it seems that the post did not load the image with the error reason. I followed the tutorial except for adding fetal fup and Hematocrit Fetus. The result failed to run, displaying ：
Simulation run failed: some variables were negative when trying to reach t=3' There are different possible reasons for this 
Solver tolerances are too high. Please reduce the absolute and relative tolerances by one order of magnitude (e.g. from 1E-9 to 1E-10) and restart the simulation.
- Some variables which are allowed to be negative were defined as non-negative.
- Model is not properly established (e.g. the kinetic should be k*[A] but was defined as k*[B] etc.)
The following variables became negative :
tm1 20mg/dJOrganism|Fetus|BloodCells|L
tm1 20mg/dJOrganism|ArterialBlood_ _UmbilicalCord|BloodCellsIL
tm120mg/dlOrganism|VenousBlood_ UmbilicalCord|Plasma|L
tm1 20mg/djOrganismjVenousBlood_ _umbilicalCordjBloodCellsIL
tm1 20mg/djOrganismjVenousBlood_ _umbilicalCordjBloodCellsIL

Note: tm1 is the added molecules start values, and L is the drug in molecules

**Reply by AndreDlm** [reply]
Thanks for the clarification. Did you try to change the solver tolerances as suggested in the error message?

When I get this error, it can be solved in most cases by reducing the absolute and relative solver tolerance by several orders of magnitude (e.g., from 1e-8 to 1e-12 or lower). You can find the solver tolerances in the Simulation under `Parameters` --> `Simulation Settings` --> `Solver Settings`

**Reply by D7abc** [reply]
Thank you very much, I'm going to try it. Best wishes to you.

**Reply by D7abc** [reply]
@AndreDlm Hi，Sorry to bother you again. When I lower the order of magnitude, it is displayed:
Error solving ODE at time t=3: Ether error test failures occurred too many times during one intermal time step, or with |h| = hmin (CV_ _ERR_ FAILURE)，What else should I do?

Best.

**Comment by AndreDlm** [comment]
Assuming that you have the default output resolution of 20 pts/h, this error most likely tells you that the numerical integrations fails already in the first step (i.e. at the simulation time of 3 min). To check this you could further increase the initial resolution, e.g. to 1000 pts/minute. 

If you get the same error at t=1/1000 min, something is wrongly implemented in the model and I suspect that some variable or parameter, e.g. molecule concentration or amount, is <0 at t=0 min. 
Without knowing the model it is quite difficult to pin down the error. To identify the erroneous parameter, you could try inactivating all new transport processes (i.e., transfer across the placenta and subsequent distribution to the fetal compartments) in the `passive transports` building block like this:
![image](https://github.com/Open-Systems-Pharmacology/Forum/assets/26541603/f71f3622-037a-4e46-acd9-03c0f05ba3bd)
You need to untick the box for all new processes (`PassiveDiffusionPlacentaBarrier`, `MassTransferPlacentaPl2VeinCordBlood`, `MassTransferPlacentaRBC2VeinCordBlood` etc).

Then, tick again the box consecutively for the different processes (starting with placental transfer, thereafter the distribution from placenta to Venous Umbilical Cord Blood plasma/RBC, then the distribution to the Fetus, and finally to Arterial Cord Blood) and simulate each time you have activated a new transport process. This should help you to identify the process that throws this error. 

Once you have identified it, you could have a look at all variables used in this process (e.g., fetal hematocrit, compartment volume etc) and check which of these parameters leads to a negative output (=simulated amount) of the process.

**Reply by D7abc** [reply]
Thank you very much for your patient answer!！

Best.

**Reply by HaiXxxxxx** [reply]
@D7abc 
Hi,
look forward to finding you well.
I'd like to ask which pregnancy model you are using as a template.
Best regards.
